server/controllers/users.js

```diff
@@ -0,0 +1,296 @@
+import * as _ from 'lodash';
+import randomstring from 'randomstring';
+import { User, Message } from '../models';
+import { transporter, helperOptions } from '../config/nodemailer';
+
+// Function to signup new users
+export const signup = (req, res) => {
+  const username = req.body.username.trim().toLowerCase();
+  const email = req.body.email.trim().toLowerCase();
```

Comment:
What if I pass in an invalid email address for example `emailaddress`? I can't see where you covered that scenario.